dae/dae/backends/impala/parquet_io.py

```diff
@@ -59,29 +61,103 @@ def __len__(self):
 
 
-class ParquetPartitionDescription(object):
-    def __init__(self, region_length, family_bin_size):
+class ParquetPartitionDescription():
+    def __init__(self,
+                 chromosomes,
+                 region_length,
+                 family_bin_size=0,
+                 coding_effect_types=[],
+                 rare_boundary=0):
+
+        self.chromosomes = chromosomes
         self.region_length = region_length
         self.family_bin_size = family_bin_size
-
-    def _evaluate_region_bin(self, summary_variant):
-        chromosome = summary_variant.ref_allele.chromosome
-        pos = summary_variant.ref_allele.position // self.region_length
-        return f'{chromosome}_{pos}'
+        self.coding_effect_types = coding_effect_types
+        self.rare_boundary = rare_boundary
+
+    def _evaluate_region_bin(self, family_variant):
+        chromosome = family_variant.ref_allele.chromosome
+        pos = family_variant.ref_allele.position // self.region_length
+        if chromosome in self.chromosomes:
+            return f'{chromosome}_{pos}'
+        else:
+            return f'other_{pos}'
 
     def _evaluate_family_bin(self, family_variant):
+        sha256 = hashlib.sha256()
         family_variant_id = family_variant.family_id
-        print(family_variant_id)
-        return hash(family_variant_id) % self.family_bin_size
-
-    def evaluate_variant_bin(self, summary_variant, family_variant):
-        return (
-            self._evaluate_region_bin(summary_variant),
-            self._evaluate_family_bin(family_variant)
-        )
-
-
-class ContinuousParquetFileWriter(object):
+        sha256.update(family_variant_id.encode())
+        digest = int(sha256.hexdigest(), 16)
+        return digest % self.family_bin_size
+
+    def _evaluate_coding_bin(self, family_variant):
+        variant_effects = set()
+        for effect in family_variant.effects:
+            variant_effects = variant_effects.union(effect.types)
+        coding_effect_types = set(self.coding_effect_types)
+
+        result = variant_effects.intersection(coding_effect_types)
+        if len(result) == 0:
+            return 0
+        else:
+            return 1
+
+    def _evaluate_frequency_bin(self, family_variant):
+        count = min(family_variant.get_attribute('af_allele_count'))
+        frequency = min(family_variant.get_attribute('af_allele_freq'))
+        if count == 1:  # Ultra rare
+            frequency_bin = 1
+        elif frequency < self.rare_boundary:  # Rare
+            frequency_bin = 2
+        else:  # Common
+            frequency_bin = 3
+
+        return frequency_bin
+
+    def evaluate_variant_filename(self, family_variant):
+        bins = (self._evaluate_region_bin(family_variant),)
+        filename = f'variants_region_bin_{bins[0]}'
+        if self.family_bin_size > 0:
+            bins += (self._evaluate_family_bin(family_variant),)
+            filename += f'_family_bin_{bins[len(bins)-1]}'
```

Comment:
don't get the just calculated bin using confusing bins indexing... store it in a variable, use it in the format and append it to the bins